src/simulate.jl

```diff
@@ -37,7 +37,98 @@ function simulate!(
     m
 end
 
-function simulate!(m::LinearMixedModel{T}; β = coef(m), σ = m.σ, θ = T[]) where {T}
+function simulate!(
+    rng::AbstractRNG,
+    m::GeneralizedLinearMixedModel{T};
+    β = coef(m),
+    σ = m.σ,
+    θ = T[],
+) where {T}
+    length(β) == length(fixef(m)) ||
+        length(β) == length(coef(m)) ||
+            throw(ArgumentError("You must specify all (non-singular) βs"))
+
+    dispersion_parameter(m) || isnan(σ) ||
+        throw(ArgumentError("You must not specify a dispersion parameter for model families without a dispersion parameter"))
+
+    β = convert(Vector{T},β)
+    σ = T(σ)
+    θ = convert(Vector{T},θ)
+
+    d = m.resp.d
+
+    if length(β) ≠ length(coef(m))
+        padding = length(coef(m)) - length(β)
+        for ii in 1:padding
+            push!(β, -0.0)
+        end
+    end
+
+    fast = (length(m.θ) == length(m.optsum.final))
+    setpar! = fast ? setθ! : setβθ!
+    params = fast ? θ : vcat(β, θ)
+    setpar!(m, params)
+
+    lm = m.LMM
+
+    # note that these m.resp.y and m.LMM.y will later be sychronized in (re)fit!()
+    # but for now we use them as distinct scratch buffers to avoid allocations
+
+    # the noise term is actually in the GLM and not the LMM part so no noise
+    # at the LMM level
+    η = fill!(m.LMM.y, zero(T))
+    y = m.resp.y
+
+    # assemble the linear predictor
+    @inbounds for trm in m.reterms             # add the unscaled random effects
+        unscaledre!(rng, η, trm)
+    end
+
+    # scale by lm.σ and add fixed-effects contribution
+    BLAS.gemv!('N', one(T), lm.X, β, lm.σ, η)
```

Comment:
Are you sure that `lm.σ` is the correct multiple here?  It just calls `sdest(lm)` which calls `varest(m)` which is based on `pwrss(m)`.  And that value is not necessarily related to any scale parameter.

Reply:
Yep, my deep dive into the GLMM code has highlighted this to me. It 'works' here because sigma is usually close to 1 for families without a dispersion parameter and I think unit scaling is correct.